packages/types/src/global-settings.ts

```diff
@@ -37,6 +37,15 @@ export const globalSettingsSchema = z.object({
 	currentApiConfigName: z.string().optional(),
 	listApiConfigMeta: z.array(providerSettingsEntrySchema).optional(),
 	pinnedApiConfigs: z.record(z.string(), z.boolean()).optional(),
+	apiConfigsCustomOrder: z
+		.array(
+			z.object({
+				id: z.string(),
+				index: z.number(),
+				pinned: z.boolean(),
```

Comment:
I left the `pinned` field here because I'm not sure what to do with the pinned configs for now. I tried doing separate reordering but that feels a bit overkill, so I reverted it and thought of leaving this here just in case we might want to do something with it in the future.